repo/test/test_repo.go

```diff
@@ -351,3 +356,150 @@ var BadStructureFile = qfs.NewMemfileBytes("badStructure.csv", []byte(`
+type MockRepo struct {
+	RootPath            string
+	IPFSPath            string
+	QriPath             string
+	TestCrypto          gen.CryptoGenerator
+	Streams             ioes.IOStreams
+	cfg                 *config.Config
+	UseMockRemoteClient bool
+}
+
+// NewMockRepo constructs the test repo and initializes everything as cheaply as possible.
+func NewMockRepo(peername, prefix string) (r MockRepo, err error) {
+	RootPath, err := ioutil.TempDir("", prefix)
+	if err != nil {
+		return r, err
+	}
+
+	// Create directory for new IPFS repo.
+	IPFSPath := filepath.Join(RootPath, "ipfs")
+	err = os.MkdirAll(IPFSPath, os.ModePerm)
+	if err != nil {
+		return r, err
+	}
+	// Build IPFS repo directory by unzipping an empty repo.
+	TestCrypto := NewTestCrypto()
+	err = TestCrypto.GenerateEmptyIpfsRepo(IPFSPath, "")
+	if err != nil {
+		return r, err
+	}
+	// Create directory for new Qri repo.
+	QriPath := filepath.Join(RootPath, "qri")
+	err = os.MkdirAll(QriPath, os.ModePerm)
+	if err != nil {
+		return r, err
+	}
+	// Create empty config.yaml into the test repo.
+	cfg := config.DefaultConfigForTesting().Copy()
+	cfg.Profile.Peername = peername
+
+	// PathFactory returns the paths for qri and ipfs roots.
+	// pathFactory := NewDirPathFactory(RootPath)
+
+	r = MockRepo{
+		RootPath:   RootPath,
+		IPFSPath:   IPFSPath,
+		QriPath:    QriPath,
+		TestCrypto: TestCrypto,
+		cfg:        cfg,
+	}
+	if err := r.WriteConfigFile(); err != nil {
+		return r, err
+	}
+	return r, nil
+}
+
+// Delete removes the test repo on disk.
+func (r *MockRepo) Delete() {
+	os.RemoveAll(r.RootPath)
+}
+
+// WriteConfigFile serializes the config file and writes it to the qri repository
+func (r *MockRepo) WriteConfigFile() error {
+	return r.cfg.WriteToFile(filepath.Join(r.QriPath, "config.yaml"))
+}
+
+// GetConfig returns the configuration for the test repo.
+func (r *MockRepo) GetConfig() *config.Config {
+	return r.cfg
+}
+
+// GetOutput returns the output from the previously executed command.
+func (r *MockRepo) GetOutput() string {
+	buffer, ok := r.Streams.Out.(*bytes.Buffer)
+	if ok {
+		return buffer.String()
+	}
+	return ""
+}
+
+// GetPathForDataset returns the path to where the index'th dataset is stored on CAFS.
+func (r *MockRepo) GetPathForDataset(index int) (string, error) {
+	dsRefs := filepath.Join(r.QriPath, "refs.fbs")
+
+	data, err := ioutil.ReadFile(dsRefs)
+	if err != nil {
+		return "", err
+	}
+
+	refs, err := repo.UnmarshalRefsFlatbuffer(data)
+	if err != nil {
+		return "", err
+	}
+
+	// If dataset doesn't exist, return an empty string for the path.
+	if len(refs) == 0 {
+		return "", err
+	}
+
+	return refs[index].Path, nil
+}
+
+// ReadBodyFromIPFS reads the body of the dataset at the given keyPath stored in CAFS.
+// TODO (b5): reprecate this rediculous function
+func (r *MockRepo) ReadBodyFromIPFS(keyPath string) (string, error) {
+	ctx := context.Background()
+	// TODO: Perhaps there is an existing cafs primitive that does this work instead?
```

Comment:
I wrote this comment originally (I'm pretty sure?) and the answer seems to be "no", so please remove it.